Adding option for allowing additional HTML extensions

fidian opened the pull request:
This closes aymericbeaumet/metalsmith-redirect#23

aymericbeaumet commented on readme.md:
```diff
@@ -296,6 +296,28 @@ metalsmith(__dirname).use(
+do not end in one of these extensions, it will not be allowed to be used as a
+redirection source.
+
+<details><summary>Example</summary>
```
You are missing the closing `</details>`.

fidian replied:
Added.

fidian commented:
I've added the missing HTML tag.

After trying for twenty minutes to adjust what you wanted and to add tests, I decided to give up and continue with my day job. The syntax checker is extremely strict and it seems like whatever I type in is wrong several ways. An auto-formatter (like prettier, which you do use) would help and I'd be willing to work a little on tests, but this amount of investment isn't worth it for the handful of lines I changed.

You're welcome to add tests, merge this PR, or close it due to lack of tests. I've invested enough time at this point and will be stopping here.

aymericbeaumet commented:
I'm using xo which is most of the time able to autofix style errors. I'm sorry you experienced it that way but it's usually a quite painless experience. If you happen to contribute again on my node repos, `yarn format` could help you.

Regarding the tests, I've added a few unit tests. Nothing fancy but just wanted to make sure that it behaved as expected: https://github.com/aymericbeaumet/metalsmith-redirect/blob/08874da4e872fe7ce5849df5a2fc194373599751/test/normalize.js#L123-L144

I've also just made a new release containing your work.

fidian commented:
Thank you so much!
